lib/backend/syncersv1/updateprocessors/profileprocessor.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/watchersyncer"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// Create a new SyncerUpdateProcessor to sync Profile data in v1 format for
+// consumption by Felix.
+func NewProfileUpdateProcessor() watchersyncer.SyncerUpdateProcessor {
```

Comment:
Apologies for not mentioning this earlier, but profiles are actually constructed of three different keys in the v1 data model.  Rules, Tags and Labels.

I think the Tags one can be ignored, but you'll need to also include Labels - where the Labels are assigned from the  LabelsToApply fields.  Take a look in compat.go in the backend which deals with handling the three separate v1 keys.

Reply:
Ok, I split it out into 3.  I left Tags empty, but it is always there as the first value.  Not sure if it might be better to have the return values be a map instead of a list of KVPairs so that its obvious which KVPairs are for which attributes, but I have the return as a list of 3 KVPairs to represent Tags, Labels, and Rules for now.